extension.js

```diff
@@ -112,9 +113,58 @@ function description(app) {
     return appName + ' → ' + app.get_title();
 }
 
-function updateHighlight(boxes) {
-  boxes.forEach(box => box.whole.remove_style_class_name('switcher-highlight'));
-  boxes.length > cursor && boxes[cursor].whole.add_style_class_name('switcher-highlight');
+function escapeChars(text) {
+    return text.replace(/[-[\]{}()*+?.,\\^$|#]/g, "\\$&");
+};
+
+function highlightText(text, query) {
+  // Don't apply highlighting if there's no input
+  if (query == "")
+    return text;
+
+  // Escape special characters in query
+  query = escapeChars(query);
+
+  // Identify substring parts to be highlighted
+  let queryExpression = "(";
+  let queries = query.split(' ');
+  let queriesLength = queries.length;
+  for (let i = 0; i < queriesLength - 1; i++) {
+    if (queries[i] != "") {
+      queryExpression += queries[i] + "|";
+    }
+  }
+  queryExpression += queries[queriesLength - 1] + ")";
+  let queryRegExp = new RegExp(queryExpression, "i");
+  let tokenRegExp = new RegExp("^" + queryExpression + "$", "i");
+
+  // Build resulting string from highlighted and non-highlighted strings
+  let result = "";
+  let tokens = text.split(queryRegExp);
+  let tokensLength = tokens.length;
+  for (let i = 0; i < tokensLength; i++) {
+    if (tokens[i].match(tokenRegExp)) {
+      result += '<span background=\"#4a90d9\" foreground=\"#ffffff\">' +
```

Comment:
I guess my main issue was how the underline was using the same color as the rest of the text, so it was creating noise instead of emphasis... The background color tried to address that, but maybe it had too much contrast?

However, the underline can be colored, and the commit I made now probably is a good compromise. What do you think?